src/main.cpp

```diff
@@ -97,8 +97,15 @@ const char *getDeviceName()
     getMacAddr(dmac);
 
     // Meshtastic_ab3c
-    static char name[20];
-    sprintf(name, "Meshtastic_%02x%02x", dmac[4], dmac[5]);
+    static char name[30];
+    // If we have an owner shortname and it is short.
+    if (owner.short_name != NULL
```

Comment:
Please have this be an alternate configuration with the default being the existing Mac based naming?